src/headers/mod.rs

```diff
@@ -0,0 +1,86 @@
+use async_trait::async_trait;
+use reqwest::header::HeaderMap;
+use std::collections::HashMap;
+use std::error::Error;
+use std::fmt::Display;
+use std::fmt::Formatter;
+use std::str::FromStr;
+
+mod lnurl_auth_jwt;
+pub use lnurl_auth_jwt::LnurlAuthJwt;
+
+/// Errors around providing headers for each VSS request.
+#[derive(Debug)]
+pub enum VssHeaderProviderError {
```

Comment:
Sounds good! This will be a change in https://github.com/lightningdevkit/vss-server/blob/main/app/src/main/proto/vss.proto right? Would you like to make the change or shall I open a PR?

Reply:
Using the `AuthError` and `InternalError` now.